### DISCUSSION TITLE
SD card init failed

### DISCUSSION THREAD

**Discussion by ly5p** [dl9rdz/rdz_ttgo_sonde#467] [Q&A]
hi, please more info about SD card

What type of FS ? Now my 32gb card if formated exfat.
connected to pins 15,35,32,33
device T-beam + ILI9225 with sd slot
![sd](https://github.com/user-attachments/assets/45aafff1-e3d4-43c9-8705-f388835ee44e)

**Comment by dl9rdz** [comment]
Needs to be formatted with a single partition (well, the first one is used if there are more than one), and MS-DOS FAT32.

ExFAT is not supported, it needs to the FAT32.

**Reply by ly5p** [reply]
formated single partition fat32, still the same :(  is there any limits for card size?

**Reply by dl9rdz** [reply]
So far I had tested only 16GB cards, but I just formatted a brand new 128G card with FAT32 and it seems to work fine...

**Reply by ly5p** [reply]
can be the problem, that i use different pins?

**Reply by dl9rdz** [reply]
The only board I have tested so far is the small TTGO Lora32 v2.1_1.6. But the pins should be fine.

**Reply by dl9rdz** [reply]
One detail that is important: SD card is initialized only on startup, i.e. you have to insert the card before starting / resetting the board. You cannot insert the board while it is already running.

**Reply by ly5p** [reply]
inserted before poweron

with putty serial port i found:
SD card init: Failed
SD Card Type: No SD card attached

**Reply by ly5p** [reply]
and the only one pin is different on connection: SD-SCK  (is this the same as CLK?) https://m.media-amazon.com/images/I/41Rm6WfTXYL._AC_UF1000,1000_QL80_.jpg

**Reply by dl9rdz** [reply]
<img width="1043" alt="Screenshot 2024-09-26 at 21 30 40" src="https://github.com/user-attachments/assets/15a03778-4938-4644-9aa8-29ec193c9b30">
I just wired up the slot of one of my test displays with your pin assignment. This looks good, it is working for me.
Yes, SCK is CLK

**Reply by dl9rdz** [reply]
(and yes, something with the background color is broken. I need to investigate this... )

**Reply by dl9rdz** [reply]
You also have to change the default button 2 pin to something different (or disable it with -1), otherwise it collides with the SD card pins. But very likely you did this was the config form will show you a warning....

**Reply by dl9rdz** [reply]
(But wait, I have to check one more thing, there is potentially one problem on the software side)

**Comment by ly5p** [comment]
with new update still the same, but colors are OK now.
